crest/Assets/Crest/Crest/Shaders/Resources/UpdateShadow.compute

```diff
@@ -124,15 +124,25 @@ void UpdateShadow(uint3 id : SV_DispatchThreadID)
 		shadowCoords.MainCameraCoords = mul(_MainCameraProjectionMatrix, wpos);
 	}
 
-	fixed2 shadow = 0.0;
+	half2 shadow = 0.0;
+	const half r_max = 0.5 - _LD_Params_Source[_LD_SliceIndex_Source].w;
 
 	// Shadow from last frame - manually implement black border
 	float3 uv_source = WorldToUV_Source(shadowCoords._WorldPosViewZ.xz, _LD_SliceIndex_Source);
 	half2 r = abs(uv_source.xy - 0.5);
-	if (max(r.x, r.y) < 0.49)
+	if (max(r.x, r.y) <= r_max)
```

Comment:
do we want similar array-bound checks to foam? Or should we be good here? Sorry, haven't thought too hard about it but just checking in-case the inconsistency isn't intentional. Could also be something we discuss outside of this change.

Reply:
ok i guess the index in shadow is fine, as _LD_SliceIndex_Source will be clamped, but needs careful checking for foam